src/network/src/ros/include/NetworkInputStream.h

```diff
@@ -0,0 +1,36 @@
+#ifndef HIVE_CONNECT_NETWORKINPUTSTREAM_H
+#define HIVE_CONNECT_NETWORKINPUTSTREAM_H
+
+#include "BaseTask.h"
+#include "INetworkInputStream.h"
+#include "Mutex.h"
+#include "logger/ILogger.h"
+#include <condition_variable>
+
+class NetworkInputStream : public INetworkInputStream {
+  public:
+    NetworkInputStream(ILogger& logger, int listeningPort);
+    ~NetworkInputStream() override;
+
+    bool receive(uint8_t* data, uint16_t length) override;
+
+    bool isReady() override;
+
+    bool start() override;
+    bool stop() override;
+
+    void acceptingClients();
+
+  private:
+    ILogger& m_logger;
+    int m_acceptingSocket;
+    int m_clientSocket;
+    bool m_hasClient;
+    int m_listeningPort;
+    bool m_runTask;
+    BaseTask<configMINIMAL_STACK_SIZE * 5> m_serverTask;
+    std::condition_variable m_conditionVar;
```

Comment:
If we are to start using a condition variable in both POSIX and FreeRTOS, we should maybe have an abstraction for it in Propolis. This is out of scope for this PR, but it would be useful.